### PR TITLE
Update the parseDate and DueDate, with test cases

### PR DESCRIPTION
**Pull request by qinguorui2001**
wait for weichang update regex, then i pass the test then merge

### CODE REVIEW

File: src/main/java/seedu/address/model/assignment/DueDate.java [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
+    private static final String DATE_VALIDATION_REGEX =
+            "^([1-9]|[0-2][0-9]|(3)[0-1])(/)([1-9]|((0)[0-9])|((1)[0-2]))(/)\\d{4}$";
+    private static final String TIME_VALIDATION_REGEX = "^(00|[0,1][0-9]|2[0-3])([0-5][0-9])$";
+    private static final String DATE_AND_TIME_VALIDATION_REGEX =
+            "^([1-9]|[0-2][0-9]|(3)[0-1])(/)([1-9]|((0)[0-9])|((1)[0-2]))(/)\\d{4}(,)"
+                    + "(00|[0,1][0-9]|2[0-3])([0-5][0-9])$";
+    private static final String LATEST_TIME_IN_DAY = "2359";
 
-    public final String value;
+    private final String value;
```
Comment (weichang18) [inline src/main/java/seedu/address/model/assignment/DueDate.java:29]: For the validation regex and value, I think it should be public to keep it consistent with the other classes, also for the value, DueDate.value might be used to display on the UI.

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 2]
```diff
@@ -121,6 +124,35 @@ public void parseEmail_validValueWithWhitespace_returnsTrimmedEmail() throws Exc
...
+    @Test
+    public void parseDate_validValueWithoutWhitespace_returnsDate() throws Exception {
+        DueDate expectedDueDate = new DueDate(VALID_DATE);
+        assertEquals(expectedDueDate, ParserUtil.parseDate(VALID_DATE));
+    }
+
+    @Test
+    public void parseDate_validValueWithoutWhitespace_returnsDateAndTime() throws Exception {
+        DueDate expectedDueDate = new DueDate(VALID_DATE);
+        assertEquals(expectedDueDate, ParserUtil.parseDate(VALID_DATE));
```
Comment (weichang18) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:146]: Maybe have a VALID_TIME = "1200" in you global variable, and for line 145 include both VALID_DATE and VALID_TIME.
Also, in the test method create a variable VALID_DATE_AND_TIME = date + , + time, and use that variable in parseDate.
So it tests when it takes in both date and time

File: src/test/java/seedu/address/model/assignment/DueDateTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        // null dueDate
+        assertThrows(NullPointerException.class, () -> DueDate.isValidDueDate(null));
+
+        // invalid dueDate
+        assertFalse(DueDate.isValidDueDate("")); // empty string
+        assertFalse(DueDate.isValidDueDate(" ")); // spaces only
+        assertFalse(DueDate.isValidDueDate("^")); // only non-alphanumeric characters
+        assertFalse(DueDate.isValidDueDate("11/11/2021*")); // contains non-alphanumeric characters
+        assertFalse(DueDate.isValidDueDate("11/11/2021,1588")); // contains the wrong time
+        assertFalse(DueDate.isValidDueDate("111/188/2021*")); // contains wrong day and month
```
Comment (weichang18) [inline src/test/java/seedu/address/model/assignment/DueDateTest.java:77]: maybe change to invalid time and invalid day and month

File: src/test/java/seedu/address/model/assignment/DueDateTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        assertFalse(DueDate.isValidDueDate("11/11/2021 1444")); // no comma but space between time and date
+        assertFalse(DueDate.isValidDueDate("11/11/2021,744")); // wrong format for time part
```
Comment (weichang18) [inline src/test/java/seedu/address/model/assignment/DueDateTest.java:79]: maybe change to "wrong format for time"

**Review by weichang18 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/assignment/DueDate.java (hunk 1 above)
Comment (qinguorui2001) [reply src/main/java/seedu/address/model/assignment/DueDate.java:29]: yes,good catch

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java (hunk 2 above)
Comment (qinguorui2001) [reply src/test/java/seedu/address/logic/parser/ParserUtilTest.java:146]: oh thx, silly mistake

File: src/test/java/seedu/address/model/assignment/DueDateTest.java (hunk 3 above)
Comment (qinguorui2001) [reply src/test/java/seedu/address/model/assignment/DueDateTest.java:77]: note,thx

File: src/test/java/seedu/address/model/assignment/DueDateTest.java [hunk 5]
```diff
@@ -61,4 +61,26 @@ public void isValidTime() {
...
+    public void isValidDueDate() {
+        // null dueDate
+        assertThrows(NullPointerException.class, () -> DueDate.isValidDueDate(null));
+
+        // invalid dueDate
+        assertFalse(DueDate.isValidDueDate("")); // empty string
+        assertFalse(DueDate.isValidDueDate(" ")); // spaces only
+        assertFalse(DueDate.isValidDueDate("^")); // only non-alphanumeric characters
+        assertFalse(DueDate.isValidDueDate("11/11/2021*")); // contains non-alphanumeric characters
+        assertFalse(DueDate.isValidDueDate("11/11/2021,1588")); // contains the invalid time
```
Comment (weichang18) [inline src/test/java/seedu/address/model/assignment/DueDateTest.java:75]: should be contains invalid time

**Review by weichang18 — CHANGES_REQUESTED** [review]
should be ok already after fixing 1 grammar mistake

**Pull request merged**